core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/util/CamelComponentUtils.java

```diff
@@ -247,12 +242,12 @@ protected static String getComponentJSon(String scheme, IProject project) {
         String json = null;
 
         try {
-			IMavenProjectFacade m2facade = MavenPlugin.getMavenProjectRegistry().create(project, new NullProgressMonitor());
-            Set<Artifact> deps = m2facade.getMavenProject(new NullProgressMonitor()).getArtifacts();
             ZipFile zf = null;
             ZipEntry ze = null;
-            for (Artifact dep : deps) {
-                zf = new ZipFile(dep.getFile());
+            IJavaProject jpr = JavaCore.create(project);
+            for (IClasspathEntry e : jpr.getResolvedClasspath(true)) {
+            	if (!e.getPath().toFile().isFile() || !e.getPath().toFile().getName().toLowerCase().endsWith(".jar")) continue;
```

Comment:
duplicated code that can be extracted to a method isJarFile and we will just call !isJarFile or directly isJarFile and avoid having a "continue" i tend to prefer having if(){} than having a continue but maybe just a matter of taste